lib/gcpspanner/browser_feature_count.go

```diff
@@ -158,50 +177,71 @@ func (c *Client) getInitialBrowserFeatureCount(
 }
 
 func createListBrowserFeatureCountMetricStatement(
-	browser string,
+	targetBrowser string,
+	targetMobileBrowser string,
 	startAt time.Time,
 	endAt time.Time,
 	pageSize int,
 	pageToken *BrowserFeatureCountCursor,
 	excludedFeatureIDs []string,
 ) spanner.Statement {
+
 	params := map[string]interface{}{
-		"browserName": browser,
-		"startAt":     startAt,
-		"endAt":       endAt,
-		"pageSize":    pageSize,
+		"targetBrowserName":       targetBrowser,
+		"targetMobileBrowserName": targetMobileBrowser,
+		"startAt":                 startAt,
+		"endAt":                   endAt,
+		"pageSize":                pageSize,
 	}
 	var pageFilter string
 	if pageToken != nil {
 		// Add filter for pagination if a page token is provided
 		pageFilter = `
-		AND BrowserReleases.ReleaseDate > @lastReleaseDate`
+		AND br.ReleaseDate > @lastReleaseDate`
 		params["lastReleaseDate"] = pageToken.LastReleaseDate
 	}
 
 	var excludedFeatureFilter string
 	if len(excludedFeatureIDs) > 0 {
 		params["excludedFeatureIDs"] = excludedFeatureIDs
-		excludedFeatureFilter = "AND bfa.WebFeatureID NOT IN UNNEST(@excludedFeatureIDs)"
+		excludedFeatureFilter = "AND cf.WebFeatureID NOT IN UNNEST(@excludedFeatureIDs)"
 	}
 
 	// Construct the query
 	// This query selects the 'ReleaseDate' and the feature counts for each release date.
 	query := fmt.Sprintf(`
+WITH CommonFeatures AS (
```

Comment:
You may want to deploy this to staging and try out the stats page there. In the beginning of this project, I saw that some CTEs took a long time. You'll see that for the main overview page, we [avoid using them](https://github.com/GoogleChrome/webstatus.dev/blob/b07fb016565fa6ff9ce20f0d425498d823331ab8/lib/gcpspanner/feature_base_query.go#L436-L540) and have a separate query for GCP spanner (but that query did horribly on the emulator so we kept the CTE one only for the emulator). But given this is a simple query, it might not be too bad. Or bad at all.

But I would be interested in the performance you see.

Reply:
To be more exact, I was seeing on the order of single digit minutes when using CTE on the other query. But I think on the order of seconds (before caching) would be okay.

Reply:
I tested this on staging, and this query seemed to load <1 second (much faster than the missing one chart), so I think we're safe here. Thanks for reminding me to check this James. 🙂